contributing.md

```diff
@@ -110,3 +162,34 @@ For C++ Styling, we follow Google's coding standard. Read more about it
 For Python Formatting, we use
 [Black](https://black.readthedocs.io/en/stable/the_black_code_style.html),
 a stricter subset of PEP8.
+
+### Keep it DRY (Don't repeat yourself)
+
+As with any software project, it's important to keep the amount of code to a minimum, so keep code duplication to a minimum!
+
+### Creating a Pull Request
+
+At any point in time you can create a pull request, so others can see your changes and give you feedback.
+Please create all pull requests to the `dev` branch.
+
+If your PR is still work in progress and not ready to be merged please add a `[WIP]` at the start of the title.
+Example:`[WIP] BoundedMean error check`
+
+***Always fill the PR template before sending a PR for better review and understanding :***
+- Description - summary of the change, the motivation, and any additionl context that will help others understand your PR
+- Affected Dependencies - List any dependencies that are required for this change.
+- **How has this been tested? (Important)**
+   - Describe the tests that you ran to verify your changes.
+   - Provide instructions so we can reproduce.
+   - List any relevant details for your test configuration.
+- Checklist - Following the code of conduct and basic guidelines.
+
+
+### Check CI and Wait for Reviews
+
+After each commit GitHub Actions will check your new code against the formatting guidelines and execute the tests to check if the test coverage is high enough.
+
+We will only merge PRs that pass the GitHub Actions checks.
+
+If your check fails, don't worry, you will still be able to make changes and make your code pass the checks.
```

Comment:
Suggestion: 
If the tests fail even after repeated try, feel free to ask someone in the slack channel

Reply:
I think we should also add a note to run black or clang-format locally before they push so that they can fix those style errors. @chinmayshah99 ?

Reply:
Definitely

Reply:
@chinmayshah99  and @codeboy5  would you like it to be something like this - 

We will only merge PRs that pass the GitHub Actions checks.

If your check fails, don't worry, you will still be able to make changes and make your code pass the checks. Feel free to ask around in the Slack channel #lib_pydp . 

Note: It is always suggested to run black  or clang format test or pytest on the code before sending a PR.

Do let me know if any other changes can be made. Or do we also give out a guideline or resource to perform the above-mentioned tests?

Reply:
Sounds good.